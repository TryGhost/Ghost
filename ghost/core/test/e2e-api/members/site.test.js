const {agentProvider, fixtureManager, matchers} = require('../../utils/e2e-framework');
const {anyEtag, stringMatching, anyContentLength} = matchers;
const assert = require('assert/strict');
const models = require('../../../core/server/models');

describe('Site Public Settings', function () {
    let membersAgent;

    before(async function () {
        membersAgent = await agentProvider.getMembersAPIAgent();
        await fixtureManager.init();
    });

    afterEach(async function () {
        await models.Settings.edit({
            key: 'members_signup_access',
            value: 'all'
        }, {context: {internal: true}});
    });

    it('Can retrieve site pubic config', async function () {
        const {body} = await membersAgent
            .get('/api/site')
            .matchBodySnapshot({
                site: {
                    version: stringMatching(/\d+\.\d+/)
                }
            })
            .matchHeaderSnapshot({
                etag: anyEtag,
                'content-length': anyContentLength
            });
        assert.equal(body.site.allow_external_signup, true);
    });

    it('Sets allow_external_signup to false when members are invite only', async function () {
        await models.Settings.edit({
            key: 'members_signup_access',
            value: 'invite'
        }, {context: {internal: true}});

        const {body} = await membersAgent
            .get('/api/site')
            .matchBodySnapshot({
                site: {
                    version: stringMatching(/\d+\.\d+/)
                }
            })
            .matchHeaderSnapshot({
                etag: anyEtag,
                'content-length': anyContentLength
            });
        assert.equal(body.site.allow_external_signup, false);
    });

    it('Sets allow_external_signup to false when portal requires checkbox', async function () {
        const {body: initialBody} = await membersAgent
            .get('/api/site');
        assert.equal(initialBody.site.allow_external_signup, true, 'This test requires the initial state to allow external signups');

        await models.Settings.edit({
            key: 'portal_signup_checkbox_required',
            value: true
        }, {context: {internal: true}});

        await models.Settings.edit({
            key: 'portal_signup_terms_html',
            value: 'I agree to the terms and conditions'
        }, {context: {internal: true}});

        const {body} = await membersAgent
            .get('/api/site')
            .matchBodySnapshot({
                site: {
                    version: stringMatching(/\d+\.\d+/)
                }
            })
            .matchHeaderSnapshot({
                etag: anyEtag,
                'content-length': anyContentLength
            });
        assert.equal(body.site.allow_external_signup, false);
    });
});
