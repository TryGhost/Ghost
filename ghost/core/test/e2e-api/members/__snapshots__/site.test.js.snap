// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site Public Settings Can retrieve site pubic config 1: [body] 1`] = `
Object {
  "site": Object {
    "accent_color": "#FF1A75",
    "allow_external_signup": true,
    "cover_image": "https://static.ghost.org/v5.0.0/images/publication-cover.jpg",
    "description": "Thoughts, stories and ideas",
    "icon": null,
    "locale": "en",
    "logo": null,
    "title": "Ghost",
    "url": "http://127.0.0.1:2369/",
    "version": StringMatching /\\\\d\\+\\\\\\.\\\\d\\+/,
  },
}
`;

exports[`Site Public Settings Can retrieve site pubic config 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "*",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": StringMatching /\\\\d\\+/,
  "content-type": "application/json; charset=utf-8",
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Encoding",
  "x-powered-by": "Express",
}
`;

exports[`Site Public Settings Sets allow_external_signup to false when members are invite only 1: [body] 1`] = `
Object {
  "site": Object {
    "accent_color": "#FF1A75",
    "allow_external_signup": false,
    "cover_image": "https://static.ghost.org/v5.0.0/images/publication-cover.jpg",
    "description": "Thoughts, stories and ideas",
    "icon": null,
    "locale": "en",
    "logo": null,
    "title": "Ghost",
    "url": "http://127.0.0.1:2369/",
    "version": StringMatching /\\\\d\\+\\\\\\.\\\\d\\+/,
  },
}
`;

exports[`Site Public Settings Sets allow_external_signup to false when members are invite only 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "*",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": StringMatching /\\\\d\\+/,
  "content-type": "application/json; charset=utf-8",
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Encoding",
  "x-powered-by": "Express",
}
`;

exports[`Site Public Settings Sets allow_external_signup to false when portal requires checkbox 1: [body] 1`] = `
Object {
  "site": Object {
    "accent_color": "#FF1A75",
    "allow_external_signup": false,
    "cover_image": "https://static.ghost.org/v5.0.0/images/publication-cover.jpg",
    "description": "Thoughts, stories and ideas",
    "icon": null,
    "locale": "en",
    "logo": null,
    "title": "Ghost",
    "url": "http://127.0.0.1:2369/",
    "version": StringMatching /\\\\d\\+\\\\\\.\\\\d\\+/,
  },
}
`;

exports[`Site Public Settings Sets allow_external_signup to false when portal requires checkbox 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "*",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": StringMatching /\\\\d\\+/,
  "content-type": "application/json; charset=utf-8",
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Encoding",
  "x-powered-by": "Express",
}
`;

exports[`Site Public Settings Sets allow_self_signup to false when members are invite only 1: [body] 1`] = `
Object {
  "site": Object {
    "accent_color": "#FF1A75",
    "allow_self_signup": false,
    "cover_image": "https://static.ghost.org/v5.0.0/images/publication-cover.jpg",
    "description": "Thoughts, stories and ideas",
    "icon": null,
    "locale": "en",
    "logo": null,
    "title": "Ghost",
    "url": "http://127.0.0.1:2369/",
    "version": StringMatching /\\\\d\\+\\\\\\.\\\\d\\+/,
  },
}
`;

exports[`Site Public Settings Sets allow_self_signup to false when members are invite only 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "*",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": StringMatching /\\\\d\\+/,
  "content-type": "application/json; charset=utf-8",
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Encoding",
  "x-powered-by": "Express",
}
`;
